### PR TITLE
Lod esmry

### PR DESCRIPTION
**Pull request by tskille**
This pr improves read efficiency for ESmry when reading from disk. 

In Equinor reservoir simulations are typically run on different machines/nodes that those being used for post processing. Hence opening results files directly from disk (not cache) is a very common use case.

This pr introduce a new result file (derived from smspec and standard unified or multiple result files). This have simulation results organized pr vector, not pr timestep as is the case with the unsmry file. The main advantage with this file, compared with the unsmry file, is that this file is more effective for load on demand (LOD). 

This file will be used by the ESmry constructor if present, if not the standard file (unsmry) will be used. 

This file can be created with a new member function in ESmry. 

```C++
bool ESmry::make_lodsmry_file()
```
or by using the small executable (make_lodsmry) included in this pr. This program can be used from the command line to make LODSMRY files for one or more simulations. 

My suggestion is that we should include a summary keyword LODSMRY which will create this file at the end of a simulation similar to how RUNSUM is requesting a RSM file. 
 
The following python script (using opm python bindings) has been used to verify performance improvements for ESmry with this pr.

```Python
import sys
import datetime

from opm.io.ecl import ESmry

if __name__ == "__main__":

    vectorList = ["TIME", "FOPT", "WOPR:P-13P", "RPR:1", "GGOR:EFB", "GEFF:EFBINJ", "GEFF:ECFBARFT"]

    start = datetime.datetime.now()
    smry1 = ESmry(sys.argv[1])
    lap1 = datetime.datetime.now()

    for key in vectorList:
        vect = smry1[key]

    lap2 = datetime.datetime.now()

    print ("runtime opening        : {}".format((lap1 - start).total_seconds()))
    print ("runtime loading vectors: {}".format((lap2 - lap1).total_seconds()))
```

Run times shown below is from testing with an RGS node (RH6) in Equinor.  As can be seen from the table below, read efficiency has been improved significantly when reading from disk. 

The best performance (by far) when reading from disk (and cashe), is when using the lodsmry file.  

 Source | File used | Runtime opening |  Runtime loading
------------ | ------------- | --------- | --------
master | unsmry | 6.022616  /  0.282899 | 12.810209   / 0.058089 
this pr | unsmry | 2.498873  / 0.296289 | 0.110763   /  0.056058
this pr | lodsmry | 0.266648  /  0.235164 | 0.04989   /  0.016956

Notice that run time is shown both for loading from disk and loading form cache, separated by a slash ( from disk / from cache)

### CODE REVIEW

**Comment by bska** [comment]
I'll review this shortly.  I'll just launch a build check first.

**Comment by bska** [comment]
jenkins build this please

File: CMakeLists_files.cmake [hunk 1]
```diff
@@ -478,6 +478,7 @@ if(ENABLE_ECL_INPUT)
     examples/opmpack.cpp
     examples/opmhash.cpp
     examples/wellgraph.cpp
+    examples/make_lodsmry.cpp
```
Comment (bska) [inline CMakeLists_files.cmake:481]: You **may** want to add `make_lodsmry` to the [`PROGRAM_SOURCE_FILES`](https://github.com/OPM/opm-common/blob/0756cd93ada466a7b96dfbe34d33103ab94981af/CMakeLists_files.cmake#L489-L493) too.  That way, it'll be copied out to the final destination during `make install` and also available in the packaged binaries.  As of right now, the `make_lodsmry` binary will only be available in the build tree, meaning a prospective user needs to build OPM-Core just to get the helpful program.

**Comment by alfbr** [comment]
> I also like the idea of having an OPM-specific keyword LODSMRY to request the simulator create this file at the end of the simulation run

Agreed. I do think it makes sense to have a short discussion first whether we want to make more out of it.  More specifically, should we make it into a performance oriented format like HDF5 while we are at it? If we are to support this in Flow, it quickly begs the question of support in post processing applications like Resinsight, and within Flow, MPI friendly file formats. In that setting we probably want a file format that we can live with for years (as changing it multiple times will quickly give incompatibilities and maintenance burden).

**Comment by joakim-hove** [comment]
I have not followed this at all; but if we are to support a new output format I will break my own principles and ask for some documentation of the format. 

In general terms I think a per vector based file format makes sense - but when we are leaving the realms standard Eclipse format files I think we should consider taking a larger step and leaving the whole `Fortran77` based legacy behind and use/design a more convenient format - like e.g. HDF5.

**Comment by tskille** [comment]
@joakim-hove, for documentation please have a look at the member function

[bool ESmry::make_lodsmry_file()](https://github.com/tskille/opm-common/blob/7434d2780d5d552e209a176e0cd2a3e3cd6ce1f9/src/opm/io/eclipse/ESmry.cpp#L954)

First array:  KEYCHECK, type CHAR, size 3 x number of summary vectors (after removing dublicate and empty vectors). This is used by the constructor to verify that the LODSMRY file is consistent with the SMSPEC file. 

Second array is RSTEP, type LOGI. size equal to number of time steps. Value is True if ministep is also a report step. 

Then there is one array for each summary vector, type REAL. Size is number of time steps and array name is V0, V1, V2 ... V(n-1)

numbering is following vector index used by the ESmry class. 

It is straight forward to calculate location for the different arrays by using the functions in EclUtil. The vector name is also checked when vectors are loaded.

The following function is used to load data. 
 
[void ESmry::Load_from_lodsmry(const std::vector<int>& keywIndVect) const](https://github.com/tskille/opm-common/blob/7434d2780d5d552e209a176e0cd2a3e3cd6ce1f9/src/opm/io/eclipse/ESmry.cpp#L546)

File: CMakeLists_files.cmake (hunk 1 above)
Comment (tskille) [reply CMakeLists_files.cmake:481]: Thanks, I have done that now.

Comment (bska) [reply CMakeLists_files.cmake:481]: > Thanks, I have done that now.

Thanks a lot.  Let's just run a final build check and then I'm ready to merge this.  The broader discussion about `LODSMRY` and HDF5 can wait until a later date.

**Comment by bska** [comment]
jenkins build this please

**Comment by bska** [comment]
>  Let's just run a final build check and then I'm ready to merge this.

It did take its sweet time to complete, but the build check came back green.  I'll merge this into master now.  Thanks a lot.

**Comment by alfbr** [comment]
>  The broader discussion about LODSMRY and HDF5 can wait until a later date.

Indeed, my intent with the comment was to initiate that discussion before implementing support for yet another file format in Flow.

**Comment by OPMUSER** [comment]
I definitely like the idea of a keyword instead of of a command line variable, but I don't like LODSMRY because we are not loading a file we are writing to a file from OPM Flow. Need to think about this, but something along the lines of SMRYHDF5 would seem more logical.

What about SQLITE database export, then one can run standard SQL commands to extract the data and run reports?

**Pull request merged**